### PR TITLE
Add recipe for SQL Advent Calendar #3 - GROUP BY

### PR DESCRIPTION
**Pull request by morsapaes**
A short example of a count aggregation, using the same source as #4 . I'm wondering if it'd be nice to use the filtering too, as a continuation of the previous example, or if it makes more sense for these to be self-contained. 🤔 

Used the SQL Client since non-windowed aggregations can't be previewed in VVP at the moment (2.3).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ The first twenty-four examples are being added as an advent calendar leading up
 
 1. [Creating Tables](recipes/01/01_create_table.md)
 2. [Filtering Data](recipes/02/02_where.md)
+3. [Aggregating Data](recipes/03/03_group_by.md)
```
Comment (knaufk) [inline README.md:15]: I was wondering whether we should start a new section for this called "Aggregations". We can also split this up later, though.

File: recipes/03/03_group_by.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# 03 Aggregating Data
+
+:bulb: This example will show how to aggregate server logs in real-time using the standard `GROUP BY` clause.
```
Comment (knaufk) [inline recipes/03/03_group_by.md:3]: Can you add the :bulb: to the other examples as well (in a separate commit)?

File: recipes/03/03_group_by.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  'fields.log_time.expression' =  '#{date.past ''15'',''5'',''SECONDS''}',
+  'fields.request_line.expression' = '#{regexify ''(GET|POST|PUT|PATCH){1}''} #{regexify ''(/search\.html|/login\.html|/prod\.html|cart\.html|/order\.html){1}''} #{regexify ''(HTTP/1\.1|HTTP/2|/HTTP/1\.0){1}''}',
+  'fields.status_code.expression' = '#{regexify ''(200|201|204|400|401|403|301){1}''}',
+  'fields.size.expression' = '#{number.numberBetween ''100'',''10000000''}'
+);
+
+SELECT status_code, 
+       COUNT(*) AS cnt_status
+FROM server_logs
+GROUP BY status_code;
```
Comment (knaufk) [inline recipes/03/03_group_by.md:36]: I am wondering if we should use two composite aggregation key, or is this a separate example? ksqlDB e.g. is weirdly concatenating joined aggregation key. Would be nice to show that Flink behaves more inline with expectations...

File: recipes/03/03_group_by.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  'fields.client_ip.expression' = '#{Internet.publicIpV4Address}',
+  'fields.client_identity.expression' =  '-',
+  'fields.userid.expression' =  '-',
+  'fields.log_time.expression' =  '#{date.past ''15'',''5'',''SECONDS''}',
+  'fields.request_line.expression' = '#{regexify ''(GET|POST|PUT|PATCH){1}''} #{regexify ''(/search\.html|/login\.html|/prod\.html|cart\.html|/order\.html){1}''} #{regexify ''(HTTP/1\.1|HTTP/2|/HTTP/1\.0){1}''}',
+  'fields.status_code.expression' = '#{regexify ''(200|201|204|400|401|403|301){1}''}',
+  'fields.size.expression' = '#{number.numberBetween ''100'',''10000000''}'
+);
+
+SELECT status_code,
```
Comment (knaufk) [inline recipes/03/03_group_by.md:33]: nit: align indentation with the other examples like https://github.com/ververica/flink-sql-cookbook/blob/master/recipes/02/02_where.md.

**Review by knaufk — APPROVED** [review]
Looks pretty good. I left a few suggestion, but no blockers.

File: README.md (hunk 1 above)
Comment (morsapaes) [reply README.md:15]: Maybe we can wait until we put out some more examples to see what grouping (😬 ) would make sense?

Comment (knaufk) [reply README.md:15]: :+1:

File: recipes/03/03_group_by.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+-- Sample user_agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.75.14 (KHTML, like Gecko) Version/7.0.3 Safari/7046A194A
+-- Regex pattern: '[^\/]+' (Match everything before '/')
+SELECT 
+  REGEXP_EXTRACT(user_agent,'[^\/]+') AS browser,
+  status_code, 
+  COUNT(*) AS cnt_status
+FROM server_logs
+GROUP BY 
+  status_code, REGEXP_EXTRACT(user_agent,'[^\/]+');
```
Comment (knaufk) [inline recipes/03/03_group_by.md:45]:
```suggestion
   REGEXP_EXTRACT(user_agent,'[^\/]+'),
  status_code;
```

**Pull request merged**